cruise-control-metrics-reporter/src/test/java/com/linkedin/kafka/cruisecontrol/metricsreporter/utils/CCContainerizedKraftCluster.java

```diff
@@ -0,0 +1,336 @@
+      String envKey = "KAFKA_" + key.toUpperCase().replace('.', '_');
+      kafkaContainer.withEnv(envKey, value);
+    }
+  }
+
+  /**
+   * Returns list of KafkaContainer broker objects within the TestContainer Kafka cluster.
+   *
+   * @return List of KafkaContainer broker objects within the TestContainer Kafka cluster.
+   */
+  public List<KafkaContainer> getBrokers() {
+    return this._brokers;
+  }
+
+  /**
+   * Returns a comma-separated list of bootstrap server addresses using plain text protocol.
+   *
+   * @return A comma-separated list of bootstrap server addresses using plain text protocol in the form {@code host:port}.
+   */
+  public String getPlainBootstrapAddress() {
+    return _brokers.stream().map(KafkaContainer::getBootstrapServers).collect(Collectors.joining(","));
+  }
+
+  /**
+   * Returns a comma-separated list of bootstrap server addresses that are reachable from container host.
+   *
+   * @return A comma-separated list of external bootstrap server addresses in the form {@code host:port}.
+   */
+  public String getExternalBootstrapAddress() {
+    return _brokers.stream()
+      .map(broker -> String.format("%s:%s", _containerHost, broker.getMappedPort(CONTAINER_EXTERNAL_LISTENER_PORT)))
+      .collect(Collectors.joining(","));
+  }
+
+  @Override
+  public void start() {
+    _brokers.parallelStream().forEach(GenericContainer::start);
+
+    Properties props = new Properties();
+    props.setProperty(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, getPlainBootstrapAddress());
+
+    try (AdminClient adminClient = AdminClient.create(props)) {
+      IntStream.range(0, _brokersNum).parallel().forEach(
+        id -> waitUntilBrokerIsReady(adminClient, id, Duration.ofSeconds(30))
+      );
+    }
+  }
+
+  @Override
+  public void stop() {
+    this._brokers.stream().parallel().forEach(GenericContainer::stop);
+  }
+
+  /**
+   * Waits until the identified broker is part of the Kafka cluster.
+   *
+   * @param adminClient The Kafka AdminClient instance.
+   * @param brokerId    The id of broker.
+   * @param timeout     The maximum duration to wait for the broker to join the cluster.
+   * @throws RuntimeException if the broker does not become part of the cluster before the timeout,
+   *                          or if the thread is interrupted while waiting.
+   */
+  public void waitUntilBrokerIsReady(AdminClient adminClient, int brokerId, Duration timeout) {
+    long deadline = System.currentTimeMillis() + timeout.toMillis();
+
+    while (System.currentTimeMillis() < deadline) {
+      try {
+        DescribeClusterResult cluster = adminClient.describeCluster();
+        Collection<Node> nodes = cluster.nodes().get();
+
+        boolean found = nodes.stream().anyMatch(node -> node.id() == brokerId);
+        if (found) {
+          return;
+        }
+
+        Thread.sleep(500);
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
```

Comment:
Do we need to treat `InterruptedException` differently than `ExecutionException`?